File: common/src/main/java/com/blackflagbin/common/facade/CommonLibrary.java
package com.blackflagbin.common.facade;

import android.app.Application;
import android.content.Context;
import android.support.annotation.NonNull;

import com.blackflagbin.common.listener.OnTokenExpiredListener;
import com.blankj.utilcode.util.Utils;

import java.util.Map;

/**
 * Created by blackflagbin on 2017/9/11.
 */

public class CommonLibrary {

    private OnTokenExpiredListener mOnTokenExpiredListener;
    private Context                mContext;
    private Class                  mApiClass;
    private String                 mBaseUrl;
    private int                    mTokenExpiredErrorCode;
    private int                    mTokenInvalidErrorCode;
    private Map<String, String>    headerMap;

    private CommonLibrary() {}

    public static CommonLibrary getInstance() {
        return InnerClass.instance;
    }

    public Context getContext() {
        return mContext;
    }

    public String getBaseUrl() {
        return mBaseUrl;
    }

    public Class getApiClass() {
        return mApiClass;
    }

    public int getTokenExpiredErrorCode() {
        return mTokenExpiredErrorCode;
    }

    public void setTokenExpiredErrorCode(int tokenExpiredErrorCode) {
        mTokenExpiredErrorCode = tokenExpiredErrorCode;
    }

    public int getTokenInvalidErrorCode() {
        return mTokenInvalidErrorCode;
    }

    public void setTokenInvalidErrorCode(int tokenInvalidErrorCode) {
        mTokenInvalidErrorCode = tokenInvalidErrorCode;
    }

    public Map<String, String> getHeaderMap() {
        return headerMap;
    }

    public void setHeaderMap(Map<String, String> headerMap) {
        this.headerMap = headerMap;
    }

    public OnTokenExpiredListener getOnTokenExpiredListener() {
        return mOnTokenExpiredListener;
    }

    public void init(
            @NonNull Application context, @NonNull String baseUrl, @NonNull Class apiClass, @NonNull OnTokenExpiredListener onTokenExpiredListener) {
        mContext = context;
        Utils.init(context);
        mBaseUrl = baseUrl;
        mApiClass = apiClass;
        mOnTokenExpiredListener = onTokenExpiredListener;
    }


    private static class InnerClass {
        static CommonLibrary instance = new CommonLibrary();
    }


}

File: kcommon/src/main/java/com/blackflagbin/kcommon/facade/CommonLibrary.kt
package com.blackflagbin.kcommon.facade

import android.app.Application
import android.content.Context
import com.blackflagbin.kcommon.listener.ErrorHandleCallBack
import com.blackflagbin.kcommon.listener.OnPageCreateListener
import com.blackflagbin.kcommon.listener.OnPageDestroyListener
import com.blankj.utilcode.util.Utils

class CommonLibrary private constructor() {

    lateinit var context: Context
    lateinit var baseUrl: String
    lateinit var apiClass: Class<*>
    lateinit var cacheClass: Class<*>
    var isDebug: Boolean = true
    var headerMap: Map<String, String>? = null
    var errorHandleMap: Map<Int, ErrorHandleCallBack>? = null
    var onPageCreateListener: OnPageCreateListener? = null
    var onPageDestroyListener: OnPageDestroyListener? = null

    fun init(
            context: Application,
            baseUrl: String,
            apiClass: Class<*>,
            cacheClass: Class<*>,
            isDebug: Boolean = true,
            headerMap: Map<String, String>? = null,
            errorHandleMap: Map<Int, ErrorHandleCallBack>? = null,
            onPageCreateListener: OnPageCreateListener? = null,
            onPageDestroyListener: OnPageDestroyListener? = null) {
        this.context = context
        Utils.init(context)
        this.baseUrl = baseUrl
        this.apiClass = apiClass
        this.cacheClass = cacheClass
        this.isDebug = isDebug
        this.headerMap = headerMap
        this.errorHandleMap = errorHandleMap
        this.onPageCreateListener = onPageCreateListener
        this.onPageDestroyListener = onPageDestroyListener
    }


    private object InnerClass {
        internal var instance = CommonLibrary()
    }

    companion object {
        val instance: CommonLibrary
            get() = InnerClass.instance
    }


}
